pkg/ctl/install/install.go

```diff
@@ -0,0 +1,435 @@
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/eks"
+	"github.com/weaveworks/eksctl/pkg/utils/kubeconfig"
+)
+
+const namespaceFileName = "namespace.yaml"
+
+// Command will create the `install` commands
+func Command(flagGrouping *cmdutils.FlagGrouping) *cobra.Command {
+	verbCmd := cmdutils.NewVerbCmd("install", "Install components in a cluster", "")
+
+	cmdutils.AddResourceCmd(flagGrouping, verbCmd, installFluxCmd)
+
+	return verbCmd
+}
+
+type installFluxOpts struct {
+	templateParams install.TemplateParameters
+	gitFluxPath    string
+	timeout        time.Duration
+	amend          bool
+}
+
+func installFluxCmd(rc *cmdutils.ResourceCmd) {
+	rc.ClusterConfig = api.NewClusterConfig()
+	rc.SetDescription(
+		"flux",
+		"Bootstrap Flux, installing it in the cluster and initializing its manifests in the Git repository",
+		"",
+		)
+	var opts installFluxOpts
+	rc.SetRunFuncWithNameArg(func() error {
+		installer, err := newFluxInstaller(rc, &opts)
+		if err != nil {
+			return err
+		}
+		return installer.run(context.Background())
+	})
+
+	rc.FlagSetGroup.InFlagSet("General", func(fs *pflag.FlagSet) {
+		fs.StringVar(&opts.templateParams.GitURL, "git-url", "",
+			"URL of the Git repository to be used by Flux, e.g. git@github.com:<your username>/flux-get-started")
+		fs.StringVar(&opts.templateParams.GitBranch, "git-branch", "master",
+			"Git branch to be used by Flux")
+		fs.StringSliceVar(&opts.templateParams.GitPaths, "git-paths", []string{},
+			"Relative paths within the Git repo for Flux to locate Kubernetes manifests")
+		fs.StringVar(&opts.templateParams.GitLabel, "git-label", "flux",
+			"Git label to keep track of Flux's sync progress; overrides both --git-sync-tag and --git-notes-ref")
+		fs.StringVar(&opts.templateParams.GitUser, "git-user", "Flux",
+			"Username to use as Git committer")
+		fs.StringVar(&opts.templateParams.GitEmail, "git-email", "",
+			"Email to use as Git committer")
+		fs.StringVar(&opts.gitFluxPath, "git-flux-subdir", "flux/",
+			"Directory within the Git repository where to commit the Flux manifests")
+		fs.DurationVar(&opts.timeout, "timeout", 20*time.Second,
+			"Timeout duration for I/O operations")
+		fs.StringVar(&opts.templateParams.Namespace, "namespace", "flux",
+			"Cluster namespace where to install flux")
+		fs.BoolVar(&opts.amend, "amend", false,
+			"Stop to manually tweak the Flux manifests before pushing them to the Git repository")
+		cmdutils.AddNameFlag(fs, rc.ClusterConfig.Metadata)
+		cmdutils.AddRegionFlag(fs, rc.ProviderConfig)
+		cmdutils.AddConfigFileFlag(fs, &rc.ClusterConfigFile)
```

Comment:
I believe `cmdutils.AddCommonFlagsForAWS` is needed here, it defines `--timeout` and other flags.

Reply:
Do we really want all of this @errordeveloper? I don't think so.
Moreover, the timeout flag says `"max wait time in any polling operations"` which is, I _believe_, unrelated to what we do in this PR, as the timeout is only used for Git-related I/O operations.

Reply:
Good point, we probably don't want to wait 30 seconds for a git command to finish :/

Reply:
Only tangential, but for a `git clone` you might